backend/controller/role.go

```diff
@@ -0,0 +1,98 @@
+package controller
+
+import (
+	"api/db"
+	"api/helpers"
+	"api/models"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func GetAllRoles(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	err, roles := helpers.GetAllTable(w, []models.Role{})
+	if err == nil {
+		helpers.JsonWriter(w, roles)
+	}
+
+	//for viewing all connections
+	//roles = []models.Role{}
+	//config.DB.Preload("Doors").Find(&roles)
+	//helpers.JsonWriter(w, roles)
```

Comment:
can we delete this?

Reply:
potentially, I found myself using it a lot for validation hence keeping it just much easier to check everything is in order by running a get all